ios/Classes/core/PMPlugin.m

```diff
@@ -38,15 +38,29 @@ - (void)onMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result {
   PMManager *manager = self.manager;
 
   if ([call.method isEqualToString:@"requestPermission"]) {
-    [PHPhotoLibrary requestAuthorization:^(PHAuthorizationStatus status) {
-        BOOL auth = PHAuthorizationStatusAuthorized == status;
-        [manager setAuth:auth];
-        if (auth) {
-          [handler reply:@1];
-        } else {
-          [handler reply:@0];
-        }
-    }];
+      if (@available(iOS 14, *)) {
```

Comment:
Just like next:
```m
#ifdef  __IPHONE_14_0
// use new api
#else
// use old api
#endif
```

Reply:
@CaiJingLong, if compatibility with Xcode 11 is important, I will change the check as suggested

Reply:
Although developers will definitely migrate to the new IDE version in the future, the old IDE cannot be given up yet.
Yes, I think this is more important at the moment.

You can try to use the macro to see if there is a problem in the XCode12+ios14 environment.

Reply:
@CaiJingLong the answer here https://stackoverflow.com/questions/3339722/how-to-check-ios-version/3339787#3339787 seems to suggest that `@available` was introduced in Xcode 9, which is is pretty old now.

Thinking about this further, I also realized that `#ifdef` is a precompiler directive, which is only evaluated at compile time. Correct me if I'm wrong, but doesn't that mean that if my compile target is ios 14+ the compile strips out everything between the `#else` and `#endif`? And Apple requires you to compile against the latest target, doesn't it? So you would compile against ios 14, but your App can run on ios 13 and below. So the check would be wrong, wouldn't it? Because it is not a runtime check anymore, but depends on your compilation target. So the code would fail on ios 13, wouldn't it?

But maybe I'm just not understanding the precompiler directives correctly.